### PR TITLE
MAR-146

### PR DESCRIPTION
**Pull request by fericsepi**
- update to how sessions are initialised during login
- update to sso client to use the token from env var
- update market access api client to handle JSONDecodeError-s
- add logout view that will flush the django session
- update context_processor to pull user info from session

### CODE REVIEW

File: utils/sessions.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        # #   first_name - STR
+        # #   last_name - STR
+        # #   email - STR
+        # #   location - STR
+        # #   internal - BOOL
+        # #   user_profile - DICT
+        # #       watchList - DICT
+        # #   permitted_applications - LIST
+        #
+        # # user_data = session['api_client'].get('whoami') or {}
```
Comment (dankingston) [inline utils/sessions.py]: Can we delete the commented code or will it be used?

File: utils/sessions.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    :param access_token: bearer token from SSO
+    :return: BOOL - whether the init process was considered successful or not
+    """
+    user_synced = False
+
+    if access_token:
+        del session['oauth_state_id']
+        session['sso_token'] = access_token
+        user_synced = sync_user(session)
+        # client = MarketAccessAPIClient(access_token)
```
Comment (dankingston) [inline utils/sessions.py]: Can we delete the commented code or will it be used?

File: utils/api_client.py [hunk 3]
```diff
@@ -28,8 +34,27 @@ def request(self, method, path, **kwargs):
         response.raise_for_status()
         return response
 
-    def get(self, path, **kwargs):
-        return self.request('get', path, **kwargs).json()
+    def get(self, path, json=True, **kwargs):
+        response = self.request('get', path, **kwargs)
+        if response.status_code is 200:
```
Comment (dankingston) [inline utils/api_client.py]: I don't think it's possible to get this far unless the status code is ok. 
`response.raise_for_status()` in the request function should raise an exception before it gets here.
You can catch it to handle what to do if the response wasn't ok e.g.
```
try:
    response = self.request('get', path, **kwargs)
except requests.exceptions.HTTPError as e:
    # Do something here?
```

Comment (fericsepi) [reply utils/api_client.py]: Ah I see what you mean.
I didn't notice that it's calling `response.raise_for_status()`
I don't think it should raise anything in the client as then the code has to use try/except but that's a diff conversation.

File: utils/sessions.py (hunk 1 above)
Comment (fericsepi) [reply utils/sessions.py]: good catch, thanks

File: utils/api_client.py (hunk 3 above)
Comment (fericsepi) [reply utils/api_client.py]: I'll leave it as is then we can revisit.

**Pull request merged**